java/ql/src/Security/CWE/CWE-780/RsaWithoutOaep.ql

```diff
@@ -0,0 +1,20 @@
+/**
+ * @name Use of RSA algorithm without OAEP
+ * @description Using RSA encryption without OAEP padding can lead to a padding oracle attack, weakening the encryption.
+ * @kind path-problem
+ * @problem.severity warning
+ * @security-severity 7.5
+ * @precision high
+ * @id java/rsa-without-oaep
+ * @tags security
+ *       external/cwe/cwe-780
+ */
+
+import java
+import semmle.code.java.security.RsaWithoutOaepQuery
+import DataFlow::PathGraph
+
+from RsaWithoutOaepConfig conf, DataFlow::Node source, DataFlow::Node sink
+where conf.hasFlow(source, sink)
+select source, source, sink,
+  "This specification is used to initialize an RSA cipher without OAEP padding $@.", sink, "here"
```

Comment:
As per the failed test: either change the query back to a `problem` kind (even if you're using a `DataFlow::Configuration`), or use `PathNode` here:

```suggestion
from RsaWithoutOaepConfig conf, DataFlow::PathNode source, DataFlow::PathNode sink
where conf.hasFlowPath(source, sink)
select source.getNode(), source, sink,
  "This specification is used to initialize an RSA cipher without OAEP padding $@.", sink.getNode(),
  "here"
```